// @metamask/delegation-abis/dist/bytecode.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var bytecode_exports = {};
__export(bytecode_exports, {
  Address: () => import_Address.bytecode,
  AllowedCalldataEnforcer: () => import_AllowedCalldataEnforcer.bytecode,
  AllowedMethodsEnforcer: () => import_AllowedMethodsEnforcer.bytecode,
  AllowedTargetsEnforcer: () => import_AllowedTargetsEnforcer.bytecode,
  ArgsEqualityCheckEnforcer: () => import_ArgsEqualityCheckEnforcer.bytecode,
  Base64: () => import_Base64.bytecode,
  Base64URL: () => import_Base64URL.bytecode,
  BitMaps: () => import_BitMaps.bytecode,
  BlockNumberEnforcer: () => import_BlockNumberEnforcer.bytecode,
  CaveatEnforcer: () => import_CaveatEnforcer.bytecode,
  Context: () => import_Context.bytecode,
  Create2: () => import_Create2.bytecode,
  DeleGatorCore: () => import_DeleGatorCore.bytecode,
  DelegationManager: () => import_DelegationManager.bytecode,
  DelegationMetaSwapAdapter: () => import_DelegationMetaSwapAdapter.bytecode,
  DeployedEnforcer: () => import_DeployedEnforcer.bytecode,
  ECDSA: () => import_ECDSA.bytecode,
  EIP712: () => import_EIP712.bytecode,
  EIP7702DeleGatorCore: () => import_EIP7702DeleGatorCore.bytecode,
  EIP7702StatelessDeleGator: () => import_EIP7702StatelessDeleGator.bytecode,
  ERC1155BalanceChangeEnforcer: () => import_ERC1155BalanceChangeEnforcer.bytecode,
  ERC1271Lib: () => import_ERC1271Lib.bytecode,
  ERC165: () => import_ERC165.bytecode,
  ERC1967Proxy: () => import_ERC1967Proxy.bytecode,
  ERC1967Utils: () => import_ERC1967Utils.bytecode,
  ERC20BalanceChangeEnforcer: () => import_ERC20BalanceChangeEnforcer.bytecode,
  ERC20PeriodTransferEnforcer: () => import_ERC20PeriodTransferEnforcer.bytecode,
  ERC20StreamingEnforcer: () => import_ERC20StreamingEnforcer.bytecode,
  ERC20TransferAmountEnforcer: () => import_ERC20TransferAmountEnforcer.bytecode,
  ERC721BalanceChangeEnforcer: () => import_ERC721BalanceChangeEnforcer.bytecode,
  ERC721TransferEnforcer: () => import_ERC721TransferEnforcer.bytecode,
  EncoderLib: () => import_EncoderLib.bytecode,
  EntryPoint: () => import_EntryPoint.bytecode,
  ExactCalldataBatchEnforcer: () => import_ExactCalldataBatchEnforcer.bytecode,
  ExactCalldataEnforcer: () => import_ExactCalldataEnforcer.bytecode,
  ExactExecutionBatchEnforcer: () => import_ExactExecutionBatchEnforcer.bytecode,
  ExactExecutionEnforcer: () => import_ExactExecutionEnforcer.bytecode,
  Exec: () => import_Exec.bytecode,
  ExecutionHelper: () => import_ExecutionHelper.bytecode,
  ExecutionLib: () => import_ExecutionLib.bytecode,
  HybridDeleGator: () => import_HybridDeleGator.bytecode,
  IAccount: () => import_IAccount.bytecode,
  IAccountExecute: () => import_IAccountExecute.bytecode,
  IAggregator: () => import_IAggregator.bytecode,
  IBeacon: () => import_IBeacon.bytecode,
  ICaveatEnforcer: () => import_ICaveatEnforcer.bytecode,
  IDeleGatorCore: () => import_IDeleGatorCore.bytecode,
  IDelegationManager: () => import_IDelegationManager.bytecode,
  IERC1155: () => import_IERC1155.bytecode,
  IERC1155Receiver: () => import_IERC1155Receiver.bytecode,
  IERC1271: () => import_IERC1271.bytecode,
  IERC1363: () => import_IERC1363.bytecode,
  IERC165: () => import_IERC165.bytecode,
  IERC173: () => import_IERC173.bytecode,
  IERC1822Proxiable: () => import_IERC1822Proxiable.bytecode,
  IERC20: () => import_IERC20.bytecode,
  IERC5267: () => import_IERC5267.bytecode,
  IERC721: () => import_IERC721.bytecode,
  IERC721Receiver: () => import_IERC721Receiver.bytecode,
  IERC7579Account: () => import_IERC7579Account.bytecode,
  IERC7821: () => import_IERC7821.bytecode,
  IEntryPoint: () => import_IEntryPoint.bytecode,
  IMetaSwap: () => import_IMetaSwap.bytecode,
  INonceManager: () => import_INonceManager.bytecode,
  IPaymaster: () => import_IPaymaster.bytecode,
  IStakeManager: () => import_IStakeManager.bytecode,
  IdEnforcer: () => import_IdEnforcer.bytecode,
  Initializable: () => import_Initializable.bytecode,
  LimitedCallsEnforcer: () => import_LimitedCallsEnforcer.bytecode,
  LogicalOrWrapperEnforcer: () => import_LogicalOrWrapperEnforcer.bytecode,
  Math: () => import_Math.bytecode,
  MessageHashUtils: () => import_MessageHashUtils.bytecode,
  MockCaveatEnforcer: () => import_MockCaveatEnforcer.bytecode,
  MockFailureCaveatEnforcer: () => import_MockFailureCaveatEnforcer.bytecode,
  ModeLib: () => import_ModeLib.bytecode,
  MultiSigDeleGator: () => import_MultiSigDeleGator.bytecode,
  MultiTokenPeriodEnforcer: () => import_MultiTokenPeriodEnforcer.bytecode,
  NativeBalanceChangeEnforcer: () => import_NativeBalanceChangeEnforcer.bytecode,
  NativeTokenPaymentEnforcer: () => import_NativeTokenPaymentEnforcer.bytecode,
  NativeTokenPeriodTransferEnforcer: () => import_NativeTokenPeriodTransferEnforcer.bytecode,
  NativeTokenStreamingEnforcer: () => import_NativeTokenStreamingEnforcer.bytecode,
  NativeTokenTransferAmountEnforcer: () => import_NativeTokenTransferAmountEnforcer.bytecode,
  NonceEnforcer: () => import_NonceEnforcer.bytecode,
  NonceManager: () => import_NonceManager.bytecode,
  Ownable: () => import_Ownable.bytecode,
  Ownable2Step: () => import_Ownable2Step.bytecode,
  OwnershipTransferEnforcer: () => import_OwnershipTransferEnforcer.bytecode,
  P256SCLVerifierLib: () => import_P256SCLVerifierLib.bytecode,
  P256VerifierLib: () => import_P256VerifierLib.bytecode,
  Panic: () => import_Panic.bytecode,
  Pausable: () => import_Pausable.bytecode,
  Proxy: () => import_Proxy.bytecode,
  RedeemerEnforcer: () => import_RedeemerEnforcer.bytecode,
  ReentrancyGuard: () => import_ReentrancyGuard.bytecode,
  SCL_RIP7212: () => import_SCL_RIP7212.bytecode,
  SCL_Wrapper: () => import_SCL_Wrapper.bytecode,
  SafeCast: () => import_SafeCast.bytecode,
  SafeERC20: () => import_SafeERC20.bytecode,
  SenderCreator: () => import_SenderCreator.bytecode,
  ShortStrings: () => import_ShortStrings.bytecode,
  SignedMath: () => import_SignedMath.bytecode,
  SimpleFactory: () => import_SimpleFactory.bytecode,
  SpecificActionERC20TransferBatchEnforcer: () => import_SpecificActionERC20TransferBatchEnforcer.bytecode,
  StakeManager: () => import_StakeManager.bytecode,
  StorageSlot: () => import_StorageSlot.bytecode,
  Strings: () => import_Strings.bytecode,
  TimestampEnforcer: () => import_TimestampEnforcer.bytecode,
  UUPSUpgradeable: () => import_UUPSUpgradeable.bytecode,
  UserOperationLib: () => import_UserOperationLib.bytecode,
  ValueLteEnforcer: () => import_ValueLteEnforcer.bytecode,
  WebAuthn: () => import_WebAuthn.bytecode
});
module.exports = __toCommonJS(bytecode_exports);
var import_Address = require('./bytecode/Address.cjs');
var import_AllowedCalldataEnforcer = require('./bytecode/AllowedCalldataEnforcer.cjs');
var import_AllowedMethodsEnforcer = require('./bytecode/AllowedMethodsEnforcer.cjs');
var import_AllowedTargetsEnforcer = require('./bytecode/AllowedTargetsEnforcer.cjs');
var import_ArgsEqualityCheckEnforcer = require('./bytecode/ArgsEqualityCheckEnforcer.cjs');
var import_Base64 = require('./bytecode/Base64.cjs');
var import_Base64URL = require('./bytecode/Base64URL.cjs');
var import_BitMaps = require('./bytecode/BitMaps.cjs');
var import_BlockNumberEnforcer = require('./bytecode/BlockNumberEnforcer.cjs');
var import_CaveatEnforcer = require('./bytecode/CaveatEnforcer.cjs');
var import_Context = require('./bytecode/Context.cjs');
var import_Create2 = require('./bytecode/Create2.cjs');
var import_DelegationManager = require('./bytecode/DelegationManager.cjs');
var import_DelegationMetaSwapAdapter = require('./bytecode/DelegationMetaSwapAdapter.cjs');
var import_DeleGatorCore = require('./bytecode/DeleGatorCore.cjs');
var import_DeployedEnforcer = require('./bytecode/DeployedEnforcer.cjs');
var import_ECDSA = require('./bytecode/ECDSA.cjs');
var import_EIP712 = require('./bytecode/EIP712.cjs');
var import_EIP7702DeleGatorCore = require('./bytecode/EIP7702DeleGatorCore.cjs');
var import_EIP7702StatelessDeleGator = require('./bytecode/EIP7702StatelessDeleGator.cjs');
var import_EncoderLib = require('./bytecode/EncoderLib.cjs');
var import_EntryPoint = require('./bytecode/EntryPoint.cjs');
var import_ERC1155BalanceChangeEnforcer = require('./bytecode/ERC1155BalanceChangeEnforcer.cjs');
var import_ERC1271Lib = require('./bytecode/ERC1271Lib.cjs');
var import_ERC165 = require('./bytecode/ERC165.cjs');
var import_ERC1967Proxy = require('./bytecode/ERC1967Proxy.cjs');
var import_ERC1967Utils = require('./bytecode/ERC1967Utils.cjs');
var import_ERC20BalanceChangeEnforcer = require('./bytecode/ERC20BalanceChangeEnforcer.cjs');
var import_ERC20PeriodTransferEnforcer = require('./bytecode/ERC20PeriodTransferEnforcer.cjs');
var import_ERC20StreamingEnforcer = require('./bytecode/ERC20StreamingEnforcer.cjs');
var import_ERC20TransferAmountEnforcer = require('./bytecode/ERC20TransferAmountEnforcer.cjs');
var import_ERC721BalanceChangeEnforcer = require('./bytecode/ERC721BalanceChangeEnforcer.cjs');
var import_ERC721TransferEnforcer = require('./bytecode/ERC721TransferEnforcer.cjs');
var import_ExactCalldataBatchEnforcer = require('./bytecode/ExactCalldataBatchEnforcer.cjs');
var import_ExactCalldataEnforcer = require('./bytecode/ExactCalldataEnforcer.cjs');
var import_ExactExecutionBatchEnforcer = require('./bytecode/ExactExecutionBatchEnforcer.cjs');
var import_ExactExecutionEnforcer = require('./bytecode/ExactExecutionEnforcer.cjs');
var import_Exec = require('./bytecode/Exec.cjs');
var import_ExecutionHelper = require('./bytecode/ExecutionHelper.cjs');
var import_ExecutionLib = require('./bytecode/ExecutionLib.cjs');
var import_HybridDeleGator = require('./bytecode/HybridDeleGator.cjs');
var import_IAccount = require('./bytecode/IAccount.cjs');
var import_IAccountExecute = require('./bytecode/IAccountExecute.cjs');
var import_IAggregator = require('./bytecode/IAggregator.cjs');
var import_IBeacon = require('./bytecode/IBeacon.cjs');
var import_ICaveatEnforcer = require('./bytecode/ICaveatEnforcer.cjs');
var import_IDelegationManager = require('./bytecode/IDelegationManager.cjs');
var import_IDeleGatorCore = require('./bytecode/IDeleGatorCore.cjs');
var import_IdEnforcer = require('./bytecode/IdEnforcer.cjs');
var import_IEntryPoint = require('./bytecode/IEntryPoint.cjs');
var import_IERC1155 = require('./bytecode/IERC1155.cjs');
var import_IERC1155Receiver = require('./bytecode/IERC1155Receiver.cjs');
var import_IERC1271 = require('./bytecode/IERC1271.cjs');
var import_IERC1363 = require('./bytecode/IERC1363.cjs');
var import_IERC165 = require('./bytecode/IERC165.cjs');
var import_IERC173 = require('./bytecode/IERC173.cjs');
var import_IERC1822Proxiable = require('./bytecode/IERC1822Proxiable.cjs');
var import_IERC20 = require('./bytecode/IERC20.cjs');
var import_IERC5267 = require('./bytecode/IERC5267.cjs');
var import_IERC721 = require('./bytecode/IERC721.cjs');
var import_IERC721Receiver = require('./bytecode/IERC721Receiver.cjs');
var import_IERC7579Account = require('./bytecode/IERC7579Account.cjs');
var import_IERC7821 = require('./bytecode/IERC7821.cjs');
var import_IMetaSwap = require('./bytecode/IMetaSwap.cjs');
var import_Initializable = require('./bytecode/Initializable.cjs');
var import_INonceManager = require('./bytecode/INonceManager.cjs');
var import_IPaymaster = require('./bytecode/IPaymaster.cjs');
var import_IStakeManager = require('./bytecode/IStakeManager.cjs');
var import_LimitedCallsEnforcer = require('./bytecode/LimitedCallsEnforcer.cjs');
var import_LogicalOrWrapperEnforcer = require('./bytecode/LogicalOrWrapperEnforcer.cjs');
var import_Math = require('./bytecode/Math.cjs');
var import_MessageHashUtils = require('./bytecode/MessageHashUtils.cjs');
var import_MockCaveatEnforcer = require('./bytecode/MockCaveatEnforcer.cjs');
var import_MockFailureCaveatEnforcer = require('./bytecode/MockFailureCaveatEnforcer.cjs');
var import_ModeLib = require('./bytecode/ModeLib.cjs');
var import_MultiSigDeleGator = require('./bytecode/MultiSigDeleGator.cjs');
var import_MultiTokenPeriodEnforcer = require('./bytecode/MultiTokenPeriodEnforcer.cjs');
var import_NativeBalanceChangeEnforcer = require('./bytecode/NativeBalanceChangeEnforcer.cjs');
var import_NativeTokenPaymentEnforcer = require('./bytecode/NativeTokenPaymentEnforcer.cjs');
var import_NativeTokenPeriodTransferEnforcer = require('./bytecode/NativeTokenPeriodTransferEnforcer.cjs');
var import_NativeTokenStreamingEnforcer = require('./bytecode/NativeTokenStreamingEnforcer.cjs');
var import_NativeTokenTransferAmountEnforcer = require('./bytecode/NativeTokenTransferAmountEnforcer.cjs');
var import_NonceEnforcer = require('./bytecode/NonceEnforcer.cjs');
var import_NonceManager = require('./bytecode/NonceManager.cjs');
var import_Ownable = require('./bytecode/Ownable.cjs');
var import_Ownable2Step = require('./bytecode/Ownable2Step.cjs');
var import_OwnershipTransferEnforcer = require('./bytecode/OwnershipTransferEnforcer.cjs');
var import_P256SCLVerifierLib = require('./bytecode/P256SCLVerifierLib.cjs');
var import_P256VerifierLib = require('./bytecode/P256VerifierLib.cjs');
var import_Panic = require('./bytecode/Panic.cjs');
var import_Pausable = require('./bytecode/Pausable.cjs');
var import_Proxy = require('./bytecode/Proxy.cjs');
var import_RedeemerEnforcer = require('./bytecode/RedeemerEnforcer.cjs');
var import_ReentrancyGuard = require('./bytecode/ReentrancyGuard.cjs');
var import_SafeCast = require('./bytecode/SafeCast.cjs');
var import_SafeERC20 = require('./bytecode/SafeERC20.cjs');
var import_SCL_RIP7212 = require('./bytecode/SCL_RIP7212.cjs');
var import_SCL_Wrapper = require('./bytecode/SCL_Wrapper.cjs');
var import_SenderCreator = require('./bytecode/SenderCreator.cjs');
var import_ShortStrings = require('./bytecode/ShortStrings.cjs');
var import_SignedMath = require('./bytecode/SignedMath.cjs');
var import_SimpleFactory = require('./bytecode/SimpleFactory.cjs');
var import_SpecificActionERC20TransferBatchEnforcer = require('./bytecode/SpecificActionERC20TransferBatchEnforcer.cjs');
var import_StakeManager = require('./bytecode/StakeManager.cjs');
var import_StorageSlot = require('./bytecode/StorageSlot.cjs');
var import_Strings = require('./bytecode/Strings.cjs');
var import_TimestampEnforcer = require('./bytecode/TimestampEnforcer.cjs');
var import_UserOperationLib = require('./bytecode/UserOperationLib.cjs');
var import_UUPSUpgradeable = require('./bytecode/UUPSUpgradeable.cjs');
var import_ValueLteEnforcer = require('./bytecode/ValueLteEnforcer.cjs');
var import_WebAuthn = require('./bytecode/WebAuthn.cjs');
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Address,
  AllowedCalldataEnforcer,
  AllowedMethodsEnforcer,
  AllowedTargetsEnforcer,
  ArgsEqualityCheckEnforcer,
  Base64,
  Base64URL,
  BitMaps,
  BlockNumberEnforcer,
  CaveatEnforcer,
  Context,
  Create2,
  DeleGatorCore,
  DelegationManager,
  DelegationMetaSwapAdapter,
  DeployedEnforcer,
  ECDSA,
  EIP712,
  EIP7702DeleGatorCore,
  EIP7702StatelessDeleGator,
  ERC1155BalanceChangeEnforcer,
  ERC1271Lib,
  ERC165,
  ERC1967Proxy,
  ERC1967Utils,
  ERC20BalanceChangeEnforcer,
  ERC20PeriodTransferEnforcer,
  ERC20StreamingEnforcer,
  ERC20TransferAmountEnforcer,
  ERC721BalanceChangeEnforcer,
  ERC721TransferEnforcer,
  EncoderLib,
  EntryPoint,
  ExactCalldataBatchEnforcer,
  ExactCalldataEnforcer,
  ExactExecutionBatchEnforcer,
  ExactExecutionEnforcer,
  Exec,
  ExecutionHelper,
  ExecutionLib,
  HybridDeleGator,
  IAccount,
  IAccountExecute,
  IAggregator,
  IBeacon,
  ICaveatEnforcer,
  IDeleGatorCore,
  IDelegationManager,
  IERC1155,
  IERC1155Receiver,
  IERC1271,
  IERC1363,
  IERC165,
  IERC173,
  IERC1822Proxiable,
  IERC20,
  IERC5267,
  IERC721,
  IERC721Receiver,
  IERC7579Account,
  IERC7821,
  IEntryPoint,
  IMetaSwap,
  INonceManager,
  IPaymaster,
  IStakeManager,
  IdEnforcer,
  Initializable,
  LimitedCallsEnforcer,
  LogicalOrWrapperEnforcer,
  Math,
  MessageHashUtils,
  MockCaveatEnforcer,
  MockFailureCaveatEnforcer,
  ModeLib,
  MultiSigDeleGator,
  MultiTokenPeriodEnforcer,
  NativeBalanceChangeEnforcer,
  NativeTokenPaymentEnforcer,
  NativeTokenPeriodTransferEnforcer,
  NativeTokenStreamingEnforcer,
  NativeTokenTransferAmountEnforcer,
  NonceEnforcer,
  NonceManager,
  Ownable,
  Ownable2Step,
  OwnershipTransferEnforcer,
  P256SCLVerifierLib,
  P256VerifierLib,
  Panic,
  Pausable,
  Proxy,
  RedeemerEnforcer,
  ReentrancyGuard,
  SCL_RIP7212,
  SCL_Wrapper,
  SafeCast,
  SafeERC20,
  SenderCreator,
  ShortStrings,
  SignedMath,
  SimpleFactory,
  SpecificActionERC20TransferBatchEnforcer,
  StakeManager,
  StorageSlot,
  Strings,
  TimestampEnforcer,
  UUPSUpgradeable,
  UserOperationLib,
  ValueLteEnforcer,
  WebAuthn
});
