// @metamask/delegation-abis/dist/index.cjs
"use strict";
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);
var index_exports = {};
__export(index_exports, {
  Address: () => import_Address.abi,
  AllowedCalldataEnforcer: () => import_AllowedCalldataEnforcer.abi,
  AllowedMethodsEnforcer: () => import_AllowedMethodsEnforcer.abi,
  AllowedTargetsEnforcer: () => import_AllowedTargetsEnforcer.abi,
  ArgsEqualityCheckEnforcer: () => import_ArgsEqualityCheckEnforcer.abi,
  Base64: () => import_Base64.abi,
  Base64URL: () => import_Base64URL.abi,
  BitMaps: () => import_BitMaps.abi,
  BlockNumberEnforcer: () => import_BlockNumberEnforcer.abi,
  CaveatEnforcer: () => import_CaveatEnforcer.abi,
  Context: () => import_Context.abi,
  Create2: () => import_Create2.abi,
  DeleGatorCore: () => import_DeleGatorCore.abi,
  DelegationManager: () => import_DelegationManager.abi,
  DelegationMetaSwapAdapter: () => import_DelegationMetaSwapAdapter.abi,
  DeployedEnforcer: () => import_DeployedEnforcer.abi,
  ECDSA: () => import_ECDSA.abi,
  EIP712: () => import_EIP712.abi,
  EIP7702DeleGatorCore: () => import_EIP7702DeleGatorCore.abi,
  EIP7702StatelessDeleGator: () => import_EIP7702StatelessDeleGator.abi,
  ERC1155BalanceChangeEnforcer: () => import_ERC1155BalanceChangeEnforcer.abi,
  ERC1271Lib: () => import_ERC1271Lib.abi,
  ERC165: () => import_ERC165.abi,
  ERC1967Proxy: () => import_ERC1967Proxy.abi,
  ERC1967Utils: () => import_ERC1967Utils.abi,
  ERC20BalanceChangeEnforcer: () => import_ERC20BalanceChangeEnforcer.abi,
  ERC20PeriodTransferEnforcer: () => import_ERC20PeriodTransferEnforcer.abi,
  ERC20StreamingEnforcer: () => import_ERC20StreamingEnforcer.abi,
  ERC20TransferAmountEnforcer: () => import_ERC20TransferAmountEnforcer.abi,
  ERC721BalanceChangeEnforcer: () => import_ERC721BalanceChangeEnforcer.abi,
  ERC721TransferEnforcer: () => import_ERC721TransferEnforcer.abi,
  EncoderLib: () => import_EncoderLib.abi,
  EntryPoint: () => import_EntryPoint.abi,
  ExactCalldataBatchEnforcer: () => import_ExactCalldataBatchEnforcer.abi,
  ExactCalldataEnforcer: () => import_ExactCalldataEnforcer.abi,
  ExactExecutionBatchEnforcer: () => import_ExactExecutionBatchEnforcer.abi,
  ExactExecutionEnforcer: () => import_ExactExecutionEnforcer.abi,
  Exec: () => import_Exec.abi,
  ExecutionHelper: () => import_ExecutionHelper.abi,
  ExecutionLib: () => import_ExecutionLib.abi,
  HybridDeleGator: () => import_HybridDeleGator.abi,
  IAccount: () => import_IAccount.abi,
  IAccountExecute: () => import_IAccountExecute.abi,
  IAggregator: () => import_IAggregator.abi,
  IBeacon: () => import_IBeacon.abi,
  ICaveatEnforcer: () => import_ICaveatEnforcer.abi,
  IDeleGatorCore: () => import_IDeleGatorCore.abi,
  IDelegationManager: () => import_IDelegationManager.abi,
  IERC1155: () => import_IERC1155.abi,
  IERC1155Receiver: () => import_IERC1155Receiver.abi,
  IERC1271: () => import_IERC1271.abi,
  IERC1363: () => import_IERC1363.abi,
  IERC165: () => import_IERC165.abi,
  IERC173: () => import_IERC173.abi,
  IERC1822Proxiable: () => import_IERC1822Proxiable.abi,
  IERC20: () => import_IERC20.abi,
  IERC5267: () => import_IERC5267.abi,
  IERC721: () => import_IERC721.abi,
  IERC721Receiver: () => import_IERC721Receiver.abi,
  IERC7579Account: () => import_IERC7579Account.abi,
  IERC7821: () => import_IERC7821.abi,
  IEntryPoint: () => import_IEntryPoint.abi,
  IMetaSwap: () => import_IMetaSwap.abi,
  INonceManager: () => import_INonceManager.abi,
  IPaymaster: () => import_IPaymaster.abi,
  IStakeManager: () => import_IStakeManager.abi,
  IdEnforcer: () => import_IdEnforcer.abi,
  Initializable: () => import_Initializable.abi,
  LimitedCallsEnforcer: () => import_LimitedCallsEnforcer.abi,
  LogicalOrWrapperEnforcer: () => import_LogicalOrWrapperEnforcer.abi,
  Math: () => import_Math.abi,
  MessageHashUtils: () => import_MessageHashUtils.abi,
  MockCaveatEnforcer: () => import_MockCaveatEnforcer.abi,
  MockFailureCaveatEnforcer: () => import_MockFailureCaveatEnforcer.abi,
  ModeLib: () => import_ModeLib.abi,
  MultiSigDeleGator: () => import_MultiSigDeleGator.abi,
  MultiTokenPeriodEnforcer: () => import_MultiTokenPeriodEnforcer.abi,
  NativeBalanceChangeEnforcer: () => import_NativeBalanceChangeEnforcer.abi,
  NativeTokenPaymentEnforcer: () => import_NativeTokenPaymentEnforcer.abi,
  NativeTokenPeriodTransferEnforcer: () => import_NativeTokenPeriodTransferEnforcer.abi,
  NativeTokenStreamingEnforcer: () => import_NativeTokenStreamingEnforcer.abi,
  NativeTokenTransferAmountEnforcer: () => import_NativeTokenTransferAmountEnforcer.abi,
  NonceEnforcer: () => import_NonceEnforcer.abi,
  NonceManager: () => import_NonceManager.abi,
  Ownable: () => import_Ownable.abi,
  Ownable2Step: () => import_Ownable2Step.abi,
  OwnershipTransferEnforcer: () => import_OwnershipTransferEnforcer.abi,
  P256SCLVerifierLib: () => import_P256SCLVerifierLib.abi,
  P256VerifierLib: () => import_P256VerifierLib.abi,
  Panic: () => import_Panic.abi,
  Pausable: () => import_Pausable.abi,
  Proxy: () => import_Proxy.abi,
  RedeemerEnforcer: () => import_RedeemerEnforcer.abi,
  ReentrancyGuard: () => import_ReentrancyGuard.abi,
  SCL_RIP7212: () => import_SCL_RIP7212.abi,
  SCL_Wrapper: () => import_SCL_Wrapper.abi,
  SafeCast: () => import_SafeCast.abi,
  SafeERC20: () => import_SafeERC20.abi,
  SenderCreator: () => import_SenderCreator.abi,
  ShortStrings: () => import_ShortStrings.abi,
  SignedMath: () => import_SignedMath.abi,
  SimpleFactory: () => import_SimpleFactory.abi,
  SpecificActionERC20TransferBatchEnforcer: () => import_SpecificActionERC20TransferBatchEnforcer.abi,
  StakeManager: () => import_StakeManager.abi,
  StorageSlot: () => import_StorageSlot.abi,
  Strings: () => import_Strings.abi,
  TimestampEnforcer: () => import_TimestampEnforcer.abi,
  UUPSUpgradeable: () => import_UUPSUpgradeable.abi,
  UserOperationLib: () => import_UserOperationLib.abi,
  ValueLteEnforcer: () => import_ValueLteEnforcer.abi,
  WebAuthn: () => import_WebAuthn.abi
});
module.exports = __toCommonJS(index_exports);
var import_Address = require('./abis/Address.cjs');
var import_AllowedCalldataEnforcer = require('./abis/AllowedCalldataEnforcer.cjs');
var import_AllowedMethodsEnforcer = require('./abis/AllowedMethodsEnforcer.cjs');
var import_AllowedTargetsEnforcer = require('./abis/AllowedTargetsEnforcer.cjs');
var import_ArgsEqualityCheckEnforcer = require('./abis/ArgsEqualityCheckEnforcer.cjs');
var import_Base64 = require('./abis/Base64.cjs');
var import_Base64URL = require('./abis/Base64URL.cjs');
var import_BitMaps = require('./abis/BitMaps.cjs');
var import_BlockNumberEnforcer = require('./abis/BlockNumberEnforcer.cjs');
var import_CaveatEnforcer = require('./abis/CaveatEnforcer.cjs');
var import_Context = require('./abis/Context.cjs');
var import_Create2 = require('./abis/Create2.cjs');
var import_DelegationManager = require('./abis/DelegationManager.cjs');
var import_DelegationMetaSwapAdapter = require('./abis/DelegationMetaSwapAdapter.cjs');
var import_DeleGatorCore = require('./abis/DeleGatorCore.cjs');
var import_DeployedEnforcer = require('./abis/DeployedEnforcer.cjs');
var import_ECDSA = require('./abis/ECDSA.cjs');
var import_EIP712 = require('./abis/EIP712.cjs');
var import_EIP7702DeleGatorCore = require('./abis/EIP7702DeleGatorCore.cjs');
var import_EIP7702StatelessDeleGator = require('./abis/EIP7702StatelessDeleGator.cjs');
var import_EncoderLib = require('./abis/EncoderLib.cjs');
var import_EntryPoint = require('./abis/EntryPoint.cjs');
var import_ERC1155BalanceChangeEnforcer = require('./abis/ERC1155BalanceChangeEnforcer.cjs');
var import_ERC1271Lib = require('./abis/ERC1271Lib.cjs');
var import_ERC165 = require('./abis/ERC165.cjs');
var import_ERC1967Proxy = require('./abis/ERC1967Proxy.cjs');
var import_ERC1967Utils = require('./abis/ERC1967Utils.cjs');
var import_ERC20BalanceChangeEnforcer = require('./abis/ERC20BalanceChangeEnforcer.cjs');
var import_ERC20PeriodTransferEnforcer = require('./abis/ERC20PeriodTransferEnforcer.cjs');
var import_ERC20StreamingEnforcer = require('./abis/ERC20StreamingEnforcer.cjs');
var import_ERC20TransferAmountEnforcer = require('./abis/ERC20TransferAmountEnforcer.cjs');
var import_ERC721BalanceChangeEnforcer = require('./abis/ERC721BalanceChangeEnforcer.cjs');
var import_ERC721TransferEnforcer = require('./abis/ERC721TransferEnforcer.cjs');
var import_ExactCalldataBatchEnforcer = require('./abis/ExactCalldataBatchEnforcer.cjs');
var import_ExactCalldataEnforcer = require('./abis/ExactCalldataEnforcer.cjs');
var import_ExactExecutionBatchEnforcer = require('./abis/ExactExecutionBatchEnforcer.cjs');
var import_ExactExecutionEnforcer = require('./abis/ExactExecutionEnforcer.cjs');
var import_Exec = require('./abis/Exec.cjs');
var import_ExecutionHelper = require('./abis/ExecutionHelper.cjs');
var import_ExecutionLib = require('./abis/ExecutionLib.cjs');
var import_HybridDeleGator = require('./abis/HybridDeleGator.cjs');
var import_IAccount = require('./abis/IAccount.cjs');
var import_IAccountExecute = require('./abis/IAccountExecute.cjs');
var import_IAggregator = require('./abis/IAggregator.cjs');
var import_IBeacon = require('./abis/IBeacon.cjs');
var import_ICaveatEnforcer = require('./abis/ICaveatEnforcer.cjs');
var import_IDelegationManager = require('./abis/IDelegationManager.cjs');
var import_IDeleGatorCore = require('./abis/IDeleGatorCore.cjs');
var import_IdEnforcer = require('./abis/IdEnforcer.cjs');
var import_IEntryPoint = require('./abis/IEntryPoint.cjs');
var import_IERC1155 = require('./abis/IERC1155.cjs');
var import_IERC1155Receiver = require('./abis/IERC1155Receiver.cjs');
var import_IERC1271 = require('./abis/IERC1271.cjs');
var import_IERC1363 = require('./abis/IERC1363.cjs');
var import_IERC165 = require('./abis/IERC165.cjs');
var import_IERC173 = require('./abis/IERC173.cjs');
var import_IERC1822Proxiable = require('./abis/IERC1822Proxiable.cjs');
var import_IERC20 = require('./abis/IERC20.cjs');
var import_IERC5267 = require('./abis/IERC5267.cjs');
var import_IERC721 = require('./abis/IERC721.cjs');
var import_IERC721Receiver = require('./abis/IERC721Receiver.cjs');
var import_IERC7579Account = require('./abis/IERC7579Account.cjs');
var import_IERC7821 = require('./abis/IERC7821.cjs');
var import_IMetaSwap = require('./abis/IMetaSwap.cjs');
var import_Initializable = require('./abis/Initializable.cjs');
var import_INonceManager = require('./abis/INonceManager.cjs');
var import_IPaymaster = require('./abis/IPaymaster.cjs');
var import_IStakeManager = require('./abis/IStakeManager.cjs');
var import_LimitedCallsEnforcer = require('./abis/LimitedCallsEnforcer.cjs');
var import_LogicalOrWrapperEnforcer = require('./abis/LogicalOrWrapperEnforcer.cjs');
var import_Math = require('./abis/Math.cjs');
var import_MessageHashUtils = require('./abis/MessageHashUtils.cjs');
var import_MockCaveatEnforcer = require('./abis/MockCaveatEnforcer.cjs');
var import_MockFailureCaveatEnforcer = require('./abis/MockFailureCaveatEnforcer.cjs');
var import_ModeLib = require('./abis/ModeLib.cjs');
var import_MultiSigDeleGator = require('./abis/MultiSigDeleGator.cjs');
var import_MultiTokenPeriodEnforcer = require('./abis/MultiTokenPeriodEnforcer.cjs');
var import_NativeBalanceChangeEnforcer = require('./abis/NativeBalanceChangeEnforcer.cjs');
var import_NativeTokenPaymentEnforcer = require('./abis/NativeTokenPaymentEnforcer.cjs');
var import_NativeTokenPeriodTransferEnforcer = require('./abis/NativeTokenPeriodTransferEnforcer.cjs');
var import_NativeTokenStreamingEnforcer = require('./abis/NativeTokenStreamingEnforcer.cjs');
var import_NativeTokenTransferAmountEnforcer = require('./abis/NativeTokenTransferAmountEnforcer.cjs');
var import_NonceEnforcer = require('./abis/NonceEnforcer.cjs');
var import_NonceManager = require('./abis/NonceManager.cjs');
var import_Ownable = require('./abis/Ownable.cjs');
var import_Ownable2Step = require('./abis/Ownable2Step.cjs');
var import_OwnershipTransferEnforcer = require('./abis/OwnershipTransferEnforcer.cjs');
var import_P256SCLVerifierLib = require('./abis/P256SCLVerifierLib.cjs');
var import_P256VerifierLib = require('./abis/P256VerifierLib.cjs');
var import_Panic = require('./abis/Panic.cjs');
var import_Pausable = require('./abis/Pausable.cjs');
var import_Proxy = require('./abis/Proxy.cjs');
var import_RedeemerEnforcer = require('./abis/RedeemerEnforcer.cjs');
var import_ReentrancyGuard = require('./abis/ReentrancyGuard.cjs');
var import_SafeCast = require('./abis/SafeCast.cjs');
var import_SafeERC20 = require('./abis/SafeERC20.cjs');
var import_SCL_RIP7212 = require('./abis/SCL_RIP7212.cjs');
var import_SCL_Wrapper = require('./abis/SCL_Wrapper.cjs');
var import_SenderCreator = require('./abis/SenderCreator.cjs');
var import_ShortStrings = require('./abis/ShortStrings.cjs');
var import_SignedMath = require('./abis/SignedMath.cjs');
var import_SimpleFactory = require('./abis/SimpleFactory.cjs');
var import_SpecificActionERC20TransferBatchEnforcer = require('./abis/SpecificActionERC20TransferBatchEnforcer.cjs');
var import_StakeManager = require('./abis/StakeManager.cjs');
var import_StorageSlot = require('./abis/StorageSlot.cjs');
var import_Strings = require('./abis/Strings.cjs');
var import_TimestampEnforcer = require('./abis/TimestampEnforcer.cjs');
var import_UserOperationLib = require('./abis/UserOperationLib.cjs');
var import_UUPSUpgradeable = require('./abis/UUPSUpgradeable.cjs');
var import_ValueLteEnforcer = require('./abis/ValueLteEnforcer.cjs');
var import_WebAuthn = require('./abis/WebAuthn.cjs');
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  Address,
  AllowedCalldataEnforcer,
  AllowedMethodsEnforcer,
  AllowedTargetsEnforcer,
  ArgsEqualityCheckEnforcer,
  Base64,
  Base64URL,
  BitMaps,
  BlockNumberEnforcer,
  CaveatEnforcer,
  Context,
  Create2,
  DeleGatorCore,
  DelegationManager,
  DelegationMetaSwapAdapter,
  DeployedEnforcer,
  ECDSA,
  EIP712,
  EIP7702DeleGatorCore,
  EIP7702StatelessDeleGator,
  ERC1155BalanceChangeEnforcer,
  ERC1271Lib,
  ERC165,
  ERC1967Proxy,
  ERC1967Utils,
  ERC20BalanceChangeEnforcer,
  ERC20PeriodTransferEnforcer,
  ERC20StreamingEnforcer,
  ERC20TransferAmountEnforcer,
  ERC721BalanceChangeEnforcer,
  ERC721TransferEnforcer,
  EncoderLib,
  EntryPoint,
  ExactCalldataBatchEnforcer,
  ExactCalldataEnforcer,
  ExactExecutionBatchEnforcer,
  ExactExecutionEnforcer,
  Exec,
  ExecutionHelper,
  ExecutionLib,
  HybridDeleGator,
  IAccount,
  IAccountExecute,
  IAggregator,
  IBeacon,
  ICaveatEnforcer,
  IDeleGatorCore,
  IDelegationManager,
  IERC1155,
  IERC1155Receiver,
  IERC1271,
  IERC1363,
  IERC165,
  IERC173,
  IERC1822Proxiable,
  IERC20,
  IERC5267,
  IERC721,
  IERC721Receiver,
  IERC7579Account,
  IERC7821,
  IEntryPoint,
  IMetaSwap,
  INonceManager,
  IPaymaster,
  IStakeManager,
  IdEnforcer,
  Initializable,
  LimitedCallsEnforcer,
  LogicalOrWrapperEnforcer,
  Math,
  MessageHashUtils,
  MockCaveatEnforcer,
  MockFailureCaveatEnforcer,
  ModeLib,
  MultiSigDeleGator,
  MultiTokenPeriodEnforcer,
  NativeBalanceChangeEnforcer,
  NativeTokenPaymentEnforcer,
  NativeTokenPeriodTransferEnforcer,
  NativeTokenStreamingEnforcer,
  NativeTokenTransferAmountEnforcer,
  NonceEnforcer,
  NonceManager,
  Ownable,
  Ownable2Step,
  OwnershipTransferEnforcer,
  P256SCLVerifierLib,
  P256VerifierLib,
  Panic,
  Pausable,
  Proxy,
  RedeemerEnforcer,
  ReentrancyGuard,
  SCL_RIP7212,
  SCL_Wrapper,
  SafeCast,
  SafeERC20,
  SenderCreator,
  ShortStrings,
  SignedMath,
  SimpleFactory,
  SpecificActionERC20TransferBatchEnforcer,
  StakeManager,
  StorageSlot,
  Strings,
  TimestampEnforcer,
  UUPSUpgradeable,
  UserOperationLib,
  ValueLteEnforcer,
  WebAuthn
});
